sdk/core/azure-core/src/main/java/com/azure/core/util/polling/implementation/LongRunningOperation.java

```diff
@@ -0,0 +1,205 @@
+package com.azure.core.util.polling.implementation;
+
+import com.azure.core.util.polling.AsyncPollResponse;
+import com.azure.core.util.polling.PollResponse;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicIntegerFieldUpdater;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * INTERNAL UTILITY CLASS.
+ * @param <PollResultT> the type of the polling response model
+ * @param <FinalResultT> the type of the final response
+ *
+ *
+ */
+// This is an internal utility method to begin a Long running operation followed by polling.
+//
+//    Using LongRunningOperation::begin we can have following signature for storage beginBlobCopy
+//
+//    Flux<AsyncPollResponse<BlobCopyStatus, BlobCopyStatus>> beginBlobCopy(param);
+//
+//    Flux<AsyncPollResponse<BlobCopyStatus, BlobCopyStatus>> pollFlux = client.beginBlobCopy(params);
+//
+//    pollFlux
+//       .takeUntil(pr -> pr.getStatus() == PollResponse.OperationStatus.IN_PROGRESS)
+//       .block();
+//
+//    pollFlux
+//       .timeout(Duration.ofMinutes(15))
+//       .flatmap(pr ->
+//           if(pr.getStatus() == PollResponse.OperationStatus.FAILED)) {
+//              return pr.cancel();
```

Comment:
do we need to cancel if failed? I expect to call cancel() in `.switchIfEmpty()`, i.e. if the poll operation times out after 15 minutes.

Reply:
you're right, the sample is wrong, will fix it.

Draft is still in progress and I probably used bad names for types like AsyncPollResponse, it's mainly to communicate the concept.

Reply:
updated sample to show case a possible use case of `cancel()`.